conbench/entities/run.py

```diff
@@ -497,6 +515,49 @@ def change_pr_number_empty_string_to_none(self, data, **kwargs):
         },
     )
 
+    @marshmallow.validates_schema
+    def validate_props(self, data, **kwargs):
+        url = data["repository"]
+        if not url.startswith("https://github.com"):
+            raise marshmallow.ValidationError(
+                f"'repository' must be a URL, starting with 'https://github.com', got `{url}`"
+            )
+
+        try:
+            urlparse(url)
+        except ValueError as exc:
+            raise marshmallow.ValidationError(
+                f"'repository' failed URL validation: `{exc}`"
+            )
```

Comment:
maybe exception-chain here?

Reply:
I think the stringified error message is the only relevant error detail from the "url parse exception".